### DISCUSSION TITLE
How to handle empty values

### DISCUSSION THREAD

**Discussion by Soffan76** [pimcore/pimcore#17463] [General]
Hi,

I'm wondering how to handle empty values in Pimcore? If a descriptor doesn´t have a value, sholuld Pimcore then store a INA, NA or null. Is this something that is common? I am not used to working with storing these, I don´t see the meaning in it.
<img width="290" alt="height" src="https://github.com/user-attachments/assets/06270620-b9f3-488f-b104-3df1afb8f5b0">

**Comment by markus-moser** [comment]
Empty values are usually stored as `null` in the database. In your screenshot it seems that you have a list of select options in the class editor, right? There it's not possible to define a real "empty" value, because the value column is always a string. You should use the default '(Empty)' select option to make the value empty:

![image](https://github.com/user-attachments/assets/44b0d9c7-66b3-42c0-9e20-6fae47875f21)


if you want to set it through the API to empty, set it to `null`:

```php
$object->setTestSelect(null)->save();

**Reply by Soffan76** [reply]
Hi Markus,
Thank you so much for your answer. 
Yes, it is a descriptor with type: Select.
Then INA, NA and null are unnecessary and default (Empty) is to be used instead?
What happens if you don´t select Empty then? What will be shown in the api?
I am new to Pimcore and really appreciate all guidance.

**Reply by markus-moser** [reply]
Yes the other options seem to be unnecessary, but it's up to your concrete installation to provide a list of values here. The list is just a "display name=>value" list. So when you select the entries the value in the value column will be shown in the API (as string).

**Reply by Soffan76** [reply]
Thank you for explaining. :)
